src/adagrams.js

```diff
@@ -1,15 +1,109 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+const LETTER_POOL = {
+  A: 9,  B: 2,  C: 2,  D: 4,  E: 12,
+  F: 2,  G: 3,  H: 2,  I: 9,  J: 1,
+  K: 1,  L: 4,  M: 2,  N: 6,  O: 8,
+  P: 2,  Q: 1,  R: 6,  S: 4,  T: 6,
+  U: 4,  V: 2,  W: 2,  X: 1,  Y: 2,
+  Z: 1,
+};
+
+const points = {
+  A: 1, E: 1, I: 1, O: 1, U: 1,
+  L: 1, N: 1, R: 1, S: 1, T: 1,
+  D: 2, G: 2,
+  B: 3, C: 3, M: 3, P: 3,
+  F: 4, H: 4, V: 4, W: 4, Y: 4,
+  K: 5,
+  J: 8, X: 8,
+  Q: 10, Z: 10
 };
 
+// wave 01
+export const drawLetters = () => {
+  const letterPoolList = [];
+  for (const[letter,count] of Object.entries(LETTER_POOL)){
+    for (let i = 0; i < count; i++){
+      letterPoolList.push(letter)
+    }
+  }
+
+  const letterList = [];
+  while (letterList.length < 10) {
+    const index = Math.floor(Math.random() * letterPoolList.length);
+    const letter = letterPoolList[index];
+    letterList.push(letter);
+    letterPoolList.splice(index, 1);
+  }
+  return letterList;
+}
+
+// wave 02
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const tempLetterBank = lettersInHand.map(letter => letter.toLowerCase());
+
+  for (let letter of input.toLowerCase()) {
```

Comment:
Since you aren't re-assigning the `letter` variable in your loop, you should use `const` instead. General guidelines for when to use `let` vs. `const`: 
- Use `const` when you do not plan to reassign the variable. This is ideal for constants, fixed references to objects or arrays, and loop variables that don’t change inside the loop body.

- Use `let` when you need to reassign the variable later, such as in loops with counters, conditionally reassigned values, or when building a value over time.

```suggestion
  for (const letter of input.toLowerCase()) {
```